android/mock204app/src/main/java/org/orbtv/mock204app/MockOrbSessionCallback.java

```diff
@@ -1624,12 +1621,20 @@ public void onRequestDialogueEnhancementOverride(int connection, String id,
             // the requested gain value shall be applied.
             appliedGain = dialogueEnhancementGain;
         }
+        boolean isValueChanged = false;
         if (MOCK_DIALOGUE_ENHANCEMENT_GAIN != appliedGain) {
             MOCK_DIALOGUE_ENHANCEMENT_GAIN = appliedGain;
-            onRequestFeatureSettingsQuery(connection, id, F_DIALOGUE_ENHANCEMENT);
+            isValueChanged = true;
         }
         mSession.onRespondDialogueEnhancementOverride(connection, id, appliedGain);
         consoleLog("Apply dialogue enhancement override, gain: " + appliedGain);
+        if (isValueChanged) {
+            consoleLog("Apply dialogue enhancement");
+            boolean isEnabled = Boolean.TRUE.equals(MOCK_ENABLE_STATUS.get(F_DIALOGUE_ENHANCEMENT));
+            notifyDialogueEnhancement(isEnabled);
+            consoleLog("Notify settings of dialogue enhancement, gain: " +
+                    (isEnabled ? MOCK_DIALOGUE_ENHANCEMENT_GAIN : 0));
+        }
```

Comment:
I'm fine with the swapping the codes
Could you check it's with the same logic with your code, or any other thing I didn't think of?
```
        mSession.onRespondDialogueEnhancementOverride(connection, id, appliedGain);
        consoleLog("Apply dialogue enhancement override, gain: " + appliedGain);
        boolean isValueChanged = false;
        if (MOCK_DIALOGUE_ENHANCEMENT_GAIN != appliedGain) {
            MOCK_DIALOGUE_ENHANCEMENT_GAIN = appliedGain;
            consoleLog("Apply dialogue enhancement");
            boolean isEnabled = Boolean.TRUE.equals(MOCK_ENABLE_STATUS.get(F_DIALOGUE_ENHANCEMENT));
            notifyDialogueEnhancement(isEnabled);
            consoleLog("Notify settings of dialogue enhancement, gain: " +
                    (isEnabled ? MOCK_DIALOGUE_ENHANCEMENT_GAIN : 0));
        }
        ```